refresh/index.js

```diff
@@ -30,10 +31,75 @@ module.exports = generators.Base.extend({
     generators.Base.apply(this, arguments);
   },
 
+  constructor: function() {
+    generators.Base.apply(this,arguments);
+
+    // Allow the user to specify where the specification file is
+    this.option('specfile', {
+      desc: 'The location of the specification file.',
+      required: false,
+      hide: true,
+      type: String
+    })
+
+    this.option('spec', {
+      desc: 'The specification in a JSON format.',
+      required: false,
+      hide: true,
+      type: String
+    })
+  },
+
   initializing: {
-    ensureInProject: actions.ensureInProject,
+
+    readSpec: function() {
+
+      if(this.options.specfile) {
+        debug('attempting to read the spec from file')
+        try {
+          this.spec = this.fs.readJSON(this.options.specfile);
+        } catch (err) {
+          this.env.error(chalk.red(err));
+        }
+      }
+
+      if(this.options.spec) {
+        debug('attempting to read the spec from cli')
+        try {
+          this.spec = JSON.parse(this.options.spec);
+        } catch (err) {
+          this.env.error(chalk.red(err));
+        }
+      }
+    },
+
+    setDestinationRootFromSpec: function() {
+      if (this.spec) {
+        // Check if we have a directory specified, else use the default one
+        this.destinationRoot(this.spec.appDir || 'swiftserver')
+      }
+    },
 
     readConfig: function() {
+      // If we have passed in the config from a previous generator - use it
+      if(this.options.config) {
+        debug('reading the config that was passed in');
+        this.config = this.options.config;
+        return;
+      }
+
+      // If we have passed a specification file with the config in
+      if(this.spec) {
+        // Check if we have specified the config in the file
+        if(this.spec.config) {
+          debug('reading the config in the specification file');
+          this.config = this.spec.config;
+          // Write the spec config to disk
+          this.fs.writeJSON(this.destinationPath('config.json'), this.config);
+          return;
+        }
+      }
```

Comment:
The whole section above in `readConfig()` is not ideal. It would be preferrable if the code path was the same whether the config came from a spec or was passed from the app generator. Perhaps we could just have the app generator construct a spec and pass that instead of passing the config?